Python/chapter03/p01_three_in_one/miguelHx.py

```diff
@@ -0,0 +1,420 @@
+"""
+Python version 3.7.0
+3.1 - Three in one
+Describe how you could use a single array to implement three stacks
+"""
+
+import copy
+import unittest
+from typing import List
+
+
+class StackNode(object):
```

Comment:
FYI: the `(object)` pattern was somewhat required in Python2. Python3 has fixed all the footguns associated with leaving it off and some folks prefer it that way!

also, if this is a data class, might be better to make it a real [`dataclasses.dataclass`](https://docs.python.org/3/library/dataclasses.html):

```python
from dataclasses import dataclass
from typing import Generic, TypeVar

T = TypeVar('T')

@dataclass
class StackNode(Generic[T]):
    data: T
    next: 'StackNode[T]'
```

dataclasses force you to use types (which is a good thing!), which help in documentation. i didn't need to add a comment to what `next` is, because the type already conveys that information, for both humans and machines.

Reply:
Updated, this is neat 👍